Mathlib/Data/Rat/Cast/Order.lean

```diff
@@ -148,6 +149,97 @@ theorem preimage_cast_uIoc (a b : ℚ) : (↑) ⁻¹' uIoc (a : K) b = uIoc a b
 end LinearOrderedField
 end Rat
 
+namespace NNRat
+
+variable {K} [LinearOrderedSemifield K]
+
+theorem cast_strictMono : StrictMono ((↑) : ℚ≥0 → K) := fun m n h => by
+  rwa [NNRat.cast_def, NNRat.cast_def, div_lt_div_iff, ← Nat.cast_mul, ← Nat.cast_mul,
+    Nat.cast_lt (α := K), ← NNRat.lt_def]
+  · simp
+  · simp
+
+@[mono]
+theorem cast_mono : Monotone ((↑) : ℚ≥0 → K) :=
+  cast_strictMono.monotone
+
+/-- Coercion from `ℚ` as an order embedding. -/
+@[simps!]
+def castOrderEmbedding : ℚ≥0 ↪o K :=
+  OrderEmbedding.ofStrictMono (↑) cast_strictMono
+
+@[simp, norm_cast]
+theorem cast_le {m n : ℚ≥0} : (m : K) ≤ n ↔ m ≤ n :=
+  castOrderEmbedding.le_iff_le
+
+@[simp, norm_cast]
+theorem cast_lt {m n : ℚ≥0} : (m : K) < n ↔ m < n :=
+  cast_strictMono.lt_iff_lt
+
+@[simp]
+theorem cast_nonpos {n : ℚ≥0} : (n : K) ≤ 0 ↔ n ≤ 0 := by
+  norm_cast
+
+@[simp]
+theorem cast_pos {n : ℚ≥0} : (0 : K) < n ↔ 0 < n := by
```

Comment:
Please call elements of `Rat`/`NNRat` `p` or `q`, not `m` or `n`

Reply:
This is consistent with the naming in the rest of the file; I'd prefer to declare this out of scope.